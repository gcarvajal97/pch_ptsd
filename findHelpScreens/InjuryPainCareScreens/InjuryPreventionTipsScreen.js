import React, { Component } from 'react';
import { View, Text, StyleSheet } from 'react-native';
import { Linking } from "expo";
import translate from '../../components/translateService';

export default class InjuryPreventionTipsScreen extends Component {
  render() {
    return (
      <View>
        <Text style={styles.paragraph}>{translate('injuryPainCare.Ch1_p1')}</Text>
        <Text style={styles.paragraph}>{translate('injuryPainCare.Ch1_p2')}{"\n"}{"\ "}
        <Text style={styles.link} onPress={()=> Linking.openURL('https://www.chop.edu/centers-programs/injury-prevention-program')}accessible
          accessible
          accessibilityLabel={translate('injuryPainCare.Ch1_accessibilityLabel')}
          accessibilityHint={translate('injuryPainCare.Ch1_accessibilityHint')}>{translate('injuryPainCare.Ch1_click_here')}</Text>
        {"\ "}{translate('injuryPainCare.Ch1_p3')}</Text>
        </View>
    );
  }
}

const styles = StyleSheet.create({
  paragraphTitle: {
    fontWeight: 'bold',
    fontSize: 20,
    alignSelf: 'stretch',
    textAlign: 'left',
    marginTop: 6,
    marginBottom: 3,
    color: '#2089DC'
  },
  paragraph: {
    textAlign: 'left',
    fontSize: 14,
    marginVertical: 2,
  },
  paragraphBold: {
    textAlign: 'left',
    fontSize: 14,
    fontWeight: 'bold',
    marginVertical: 2,
    marginLeft: 10
  },
  link: {
    fontSize: 14,
    color:'#2089DC',
    textDecorationLine: 'underline',
  }
});