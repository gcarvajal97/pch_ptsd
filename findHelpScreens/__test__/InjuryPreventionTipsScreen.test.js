import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import React from 'react';
import InjuryPreventionTipsScreen from '../InjuryPainCareScreens/InjuryPreventionTipsScreen';
import renderer from 'react-test-renderer';
import {Linking} from 'expo';

configure({ adapter: new Adapter() });
jest.mock('expo', () => ({
    Linking: {
        openURL: jest.fn()
    }
}));


describe("InjuryPreventionTipsScreen", () => {
    afterEach(() => {
        jest.clearAllMocks();
      });

    it("renders correctly", () => {
        expect(renderer.create(<InjuryPreventionTipsScreen/>).toJSON()).toMatchSnapshot();
    });

    it("clicks external link, opens correct website", () => {
        const wrapper = shallow(<InjuryPreventionTipsScreen/>);
        wrapper.findWhere((node) => node.prop('testID') === 'hyperlink').props().onPress();
        expect(Linking.openURL).toHaveBeenCalledWith('https://www.chop.edu/centers-programs/injury-prevention-program');
    })
})