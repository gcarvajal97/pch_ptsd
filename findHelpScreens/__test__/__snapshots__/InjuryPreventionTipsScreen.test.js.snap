// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InjuryPreventionTipsScreen renders correctly 1`] = `
<View>
  <Text
    style={
      Object {
        "fontSize": 14,
        "marginVertical": 2,
        "textAlign": "left",
      }
    }
  >
    Each year, one in every four children is treated by a doctor because of an injury such as a fall from a bicycle or an animal bite. Some of these injuries are minor, but many of them are very serious.
  </Text>
  <Text
    style={
      Object {
        "fontSize": 14,
        "marginVertical": 2,
        "textAlign": "left",
      }
    }
  >
    You can’t always stop your children from getting bumps and bruises but you can reduce the chances that they’ll be seriously injured by taking safety precautions such as always buckling up in motor vehicles.
    

     
    <Text
      accessibilityHint="Navigates to a website about safety tips"
      accessibilityLabel="Click here for safety tips"
      accessible={true}
      onPress={[Function]}
      style={
        Object {
          "color": "#0077C8",
          "fontSize": 14,
          "textDecorationLine": "underline",
        }
      }
      testID="hyperlink"
    >
      Click here
    </Text>
     
    to find more safety tips on topics such as child passenger safety, bicycle safety, dog bites, falls, pedestrian safety, fire safety, water safety and more.
  </Text>
</View>
`;
